englishLearnBot-stage3

Name707 opened the pull request:
Добавлено меню команд (учить слова, статистика или выход). Реализована команда вывода правильных ответов - количество + %

Name707 committed:
englishLearnBot-stage3
Добавлено меню команд (учить слова, статистика или выход). Реализована команда вывода правильных ответов - количество + %

ievetrov commented on src/main/kotlin/Main.kt:
```diff
@@ -13,18 +14,27 @@ fun main() {
+        when (readln().toInt()) {
+            1 -> println("Выбрали команду 1")
+            2 -> println(
+                "Выучено ${dictionary.sumOf { it.correctAnswersCount }} из ${dictionary.size} слов | ${
```
Функция sumOf не подходит. Словарь состоит из набора слов. А слово содержит:
- слово на английском
- перевод
- **количество правильных ответов для данного слова**
При изучении слов ведь можно угадать, поэтому мы считаем выученным то слово, которое было корректно выбрано 3 раза. 

Обрати внимание на эту часть инструкции:
> Теперь в when для пункта 2 добавим нужный функционал. Extension функция filter для словаря отфильтрует значения, у которых correctAnswersCount больше или равно 3. Показываем размер этого списка. Там же делаем println(), в котором пишем "сколько" слов выучено из "общего количества" и сколько это будет "в процентах". Высчитываем эти переменные и добавляем в строку.

Name707 committed:
englishLearnBot-stage3
Исправлено замечание. Получение количества выученных слов через функцию filter {}.

ievetrov commented on src/main/kotlin/Main.kt:
```diff
@@ -13,18 +14,27 @@ fun main() {
+            2 -> println(
+                "Список содержит ${dictionary.filter { it.correctAnswersCount >= 3 }.size} выученных слов." +
+                        "\nВыучено ${dictionary.filter { it.correctAnswersCount >= 3 }.size} из ${dictionary.size} слов | " +
+                        "${(((dictionary.filter { it.correctAnswersCount >= 3 }.size).toDouble() / dictionary.size) * 100).roundToInt() }%"
```
Отлично. А теперь [избавимся от дублирования](https://ru.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself) и можно вливать.

Кстати хороший пример, почему не стоит дублировать код. В последующих частях курсовой цифру 3 мы будем выносить в параметры, и вполне возможно, из какого-то filter'а ты забудешь вынести 3-ку. Убрав дублирование сейчас ты **в три раза сократишь риск ошибки**.

Name707 committed:
englishLearnBot-stage3
Исправлено замечание - "повтор кода"
